PubHub/PubHub.Common/Services/AuthorService.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System.Diagnostics;
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using PubHub.Common.ApiService;
+using PubHub.Common.Models.Authors;
+using static PubHub.Common.IntegrityConstants;
+
+namespace PubHub.Common.Services
+{
+    public class AuthorService : ServiceRoot, IAuthorService
+    {
+#pragma warning disable IDE0270 // Use coalesce expression
+        private readonly JsonSerializerOptions _serializerOptions;
+
+        internal AuthorService(IHttpClientFactory clientFactory, string clientName) : base(clientFactory, clientName)
+        {
+            _serializerOptions = new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                WriteIndented = true,
+            };
+        }
+
+        /// <summary>
+        /// Calls the API enpoint to retrieve all authors through the <see cref="AuthorInfoModel"/>.
+        /// </summary>
+        /// <returns>A list of <see cref="AuthorInfoModel"/></returns>
+        public async Task<List<AuthorInfoModel>> GetAuthors()
+        {
+            try
+            {
+                HttpResponseMessage response = await Client.GetAsync($"authors");
+                string content = await response.Content.ReadAsStringAsync();
+
+                if (!response.IsSuccessStatusCode)
+                {
+                    ErrorResponse? errorResponse = JsonSerializer.Deserialize<ErrorResponse>(content, _serializerOptions);
+                    if (errorResponse == null)
+                        throw new NullReferenceException($"Unable to handle the Error response, status code: {response.StatusCode}");
+
+                    throw new Exception($"Unable to retrieve information: {errorResponse!.Detail}");
+                }
+
+                List<AuthorInfoModel>? authorInfoModel = JsonSerializer.Deserialize<List<AuthorInfoModel>>(content, _serializerOptions);
+
+                if (authorInfoModel == null)
+                    throw new NullReferenceException($"Unable to map the request over to the client.");
+
+                return authorInfoModel!;
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine("Unable to get authors:", ex.Message);
+                return null!;
```

Comment:
Wouldn't we want to return an empty list here instead? 🤔 (I know it's been done like this in the other services. I just came to think of it now 🙃)

Reply:
I completely missed this!
Good spot.

We should never return null for a list.
The API itself doesn't even do that.

Reply:
I can change it here and then make an issue to update the other services to do this, if @ZhakalenDk also agrees to do it like this - I am fine with whatever. 😊

Reply:
Absolutely. It's bad practise to rreturn null whne working with collections (:

Reply:
I changed it - can either one of you review if this is what we want instead? 😊